server/src/main/java/keywhiz/service/resources/BatchSecretDeliveryResource.java

```diff
@@ -130,9 +131,11 @@ public BatchSecretDeliveryResource(@Readonly SecretController secretController,
     logger.info("Client {} granted access to {}.", client.getName(), clientAccessibleSecrets.stream().map(s -> s.name()).collect(toList()));
     try {
       // This is only possible if all secrets are both existing AND accessible to the client
-      return existingSecrets.stream()
+      List<SecretDeliveryResponse> secrets = existingSecrets.stream()
               .map(SecretDeliveryResponse::fromSecret)
               .collect(toList());
+      setTag("nSecrets", secrets.size());
+      return secrets;
```

Comment:
For methods which occasionally take in the hundreds of milliseconds, presumably due to larger result set sizes, I added a record of just how many things we found.

It's worth noting that this batching method has poor behavior in that if X secrets are found then X sequential calls are made to the HSM. We should look into a batching API at the HSM level.

Reply:
Should we make a TODO with a ticket? Then again, I'm not sure if we actually take action on TODOs anyway...

Reply:
I created an issue and linked it to the original issue for this task so that it wouldn't get lost.